### PR TITLE
add unit preview pane hp xp mp tooltips

### PR DESCRIPTION
**Pull request by gfgtdf**
wip dontmerge

### CODE REVIEW

**Comment by CelticMinstrel** [comment]
I'd say these should probably be squashed into one commit. Apart from that, the code looks okay to me, though I didn't look really closely.

**Comment by Vultraz** [comment]
You can set `use_markup = true` in the WML now.

**Comment by gfgtdf** [comment]
> You can set use_markup = true in the WML now.

Hmm i could, but since the c++ code always generates code that uses markup mit might be bette to enforce use_makrup=yes on the c++ side too.

**Comment by Vultraz** [comment]
Also why did you add an include for help_impl.hpp?

**Comment by gfgtdf** [comment]
> Also why did you add an include for help_impl.hpp?

I use `help::load_terrain_types_data()`

**Comment by Vultraz** [comment]
Hm, ok.

File: src/gui/widgets/unit_preview_pane.cpp [hunk 1]
```diff
@@ -98,6 +101,93 @@ static inline tree_view_node& add_name_tree_node(tree_view_node& header_node, co
...
+	for (const utils::string_map::value_type &resist : res)
+	{
+		std::ostringstream line;
+		line << translation::gettext(resist.first.c_str()) << ": ";
+		// Some units have different resistances when attacking or defending.
+		int res_att = 100 - get(resist.first, true);
+		int res_def = 100 - get(resist.first, false);
+
+		if (res_att == res_def) {
+			line << "<span foreground=\"" << unit_helper::resistance_color(res_def) << "\">" << utils::signed_percent(res_def) << "</span>" << '\n';
```
Comment (Vultraz) [inline src/gui/widgets/unit_preview_pane.cpp]: For the record, we usually use <span color=>

Comment (gfgtdf) [inline src/gui/widgets/unit_preview_pane.cpp]: i just copied that code from https://github.com/wesnoth/wesnoth/blob/master/src/reports.cpp#L432

**Comment by gfgtdf** [comment]
testing upload 
[DT-Der_Hafen_von_Schwarzwasser.gz](https://github.com/wesnoth/wesnoth/files/640277/DT-Der_Hafen_von_Schwarzwasser.gz)

**Pull request merged**